clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -687,7 +687,7 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
         throwIfProducerClosed();
         TransactionalRequestResult result = transactionManager.sendOffsetsToTransaction(offsets, groupMetadata);
         sender.wakeup();
-        result.await();
+        result.await(maxBlockTimeMs, TimeUnit.MILLISECONDS);
```

Comment:
javadoc for this method should be updated as well.

Reply:
I wrote some description related to TimeoutException and InterruptedException